### DISCUSSION TITLE
MenuButton styling

### DISCUSSION THREAD

**Discussion by spidertwin2** [chakra-ui/chakra-ui#2920] [Q&A] [answered]
Namaste,

MenuButton is sometimes not covering entire text and icon on hover/click, please see images below. 
![img1](https://user-images.githubusercontent.com/29196888/103397978-8ed1f480-4b60-11eb-8f03-7732c437179a.png)
![img2](https://user-images.githubusercontent.com/29196888/103397987-91cce500-4b60-11eb-96a9-3ade897580c6.png)

LoggedInNavbar.js
```
const LoggedInNavbar = () => {
  const router = useRouter();
  const [isLargerThan768] = useMediaQuery("(min-width: 768px)");

  return (
    <Flex
      direction="row"
      p="3% 10%"
      justify="space-between"
      color={useColorModeValue("gray.800", "white")}
      position="relative"
      zIndex="dropdown"
    >
      <Flex textStyle="brandHeading" m="auto 0">
        <Link href="/">
          <a>App</a>
        </Link>
      </Flex>
      <Flex justify="space-between">
        <DarkModeSwitch />
        <Menu>
          {isLargerThan768 ? (
            <MenuButton
              variant="ghost"
              as={Button}
              rightIcon={<ChevronDownIcon />}
              p="0 25px"
            >
              MyAccount
            </MenuButton>
          ) : (
            <MenuButton variant="ghost" p="3%" m="auto" as={Button}>
              <HamburgerIcon />
            </MenuButton>
          )}
          <MenuList>
            <MenuItem onClick={() => router.push("/user/profile")}>
              Profile
            </MenuItem>
            <MenuItem onClick={() => router.push("/user/community")}>
              Community
            </MenuItem>
            <MenuItem onClick={() => router.push("/user/resources")}>
              Resources
            </MenuItem>
            <MenuItem onClick={() => router.push("/user/discussion")}>
              Discussion
            </MenuItem>
            <MenuItem onClick={() => router.push("/user/stats")}>
              Stats
            </MenuItem>
            <MenuDivider />
            <MenuItem onClick={() => router.push("/user/settings")}>
              Settings
            </MenuItem>
            <MenuItem onClick={() => router.push("/user/signout")}>
              SignOut
            </MenuItem>
          </MenuList>
        </Menu>
      </Flex>
    </Flex>
  );
};
```
Also, these errors are being thrown when I resize my window to check for responsive styling-
![error1](https://user-images.githubusercontent.com/29196888/103398256-ecb30c00-4b61-11eb-86ef-d7052fd59033.png)
![error2](https://user-images.githubusercontent.com/29196888/103398259-f0469300-4b61-11eb-9a22-66b4c5cf4932.png)
![error3](https://user-images.githubusercontent.com/29196888/103398261-f2a8ed00-4b61-11eb-9d13-e6e1f37c9f59.png)

Thank you :smiley:

**Comment by chr-ge** [comment]
I'm not seeing any issues when I run your code in a [codesandbox](https://codesandbox.io/s/cocky-framework-29fgz?file=/src/index.js). Not sure what is causing your issues.

**Reply by spidertwin2** [reply]
Surprising, could it be a React hydration issue?

I received the following warning - 
`Text content did not match. Server: "" Client: "MyAccount"`

I stumbled on [this](https://lihautan.com/hydrating-text-content/), but was not sure what I should change in my code to resolve it.  If it is not a bother, please have a read.

Best wishes for the new year :christmas_tree:

**Comment by spidertwin2** [comment, marked as answer]
I realized this error was caused by a custom useWindowResize hook in my application, that returned the current screen size.  I switched to `useBreakpointValue` and then the error was gone.
